File: app/src/main/java/ru/vpcb/test/map/di/AppModule.java
package ru.vpcb.test.map.di;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import ru.vpcb.test.map.data.repository.FirebaseUserRepository;
import ru.vpcb.test.map.data.repository.UserRepository;
import ru.vpcb.test.map.executors.AppExecutor;
import ru.vpcb.test.map.executors.IAppExecutors;
import ru.vpcb.test.map.map.GeneralMapFragment;
import ru.vpcb.test.map.map.GoogleMapPresenter;
import ru.vpcb.test.map.map.MapFragment;
import ru.vpcb.test.map.map.MapMvpPresenter;

@Module
public class AppModule {

    @Provides
    @Singleton
    IAppExecutors provideAppExecutors() {
        return new AppExecutor();
    }

    @Provides
    @Singleton
    UserRepository provideFirebaseUserRepository(IAppExecutors appExecutors) {
        return new FirebaseUserRepository(appExecutors);
    }


    @Provides
    @Singleton
    MapMvpPresenter provideMapMvpPresenter() {
        return new GoogleMapPresenter();
    }

    @Provides
    @Singleton
    MapFragment provideMapFragment() {
        return new GeneralMapFragment();
    }

}

File: app/src/main/java/ru/vpcb/map/notes/AppModule.kt
package ru.vpcb.map.notes

import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import ru.vpcb.map.notes.data.remote.FirebaseRepo
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
object AppModule {

    @Provides
    @Singleton
    fun provideFBUserRepo() = FirebaseRepo()
}